### PR TITLE
gnonograms: init at 1.4.5

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122625` at 23403cb1 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>gnonograms</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/meson-cmake.md">meson-cmake</a>

Meson uses CMake as a fallback dependency resolution method and it likely is not necessary here. The message about cmake not being found is purely informational.

Near pkgs/games/gnonograms/default.nix:34:3:
```
   |
34 |   nativeBuildInputs = [
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/games/gnonograms/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  src = fetchFromGitHub {
+    owner = "jeremypw";
+    repo = "gnonograms";
+    rev = "v${version}";
+    sha256 = "1ly3inp6dvjrixdysz5hdfwlhbs49ks0lf8062z2iq6gaf8ivkb2";
+  };
+
+  # Don't update mime/desktop/icon/schemas caches
+  postPatch = ''
+    sed -i '/post_install/d' meson.build
```
Comment (jtojnar) [inline pkgs/games/gnonograms/default.nix:30]: I am surprised it will run without schemas compiled.

Comment (jtojnar) [reply pkgs/games/gnonograms/default.nix:30]: Typically, you would do `patchShebangs meson/post_install.py` in `postPatch` to be able to run the script.

Comment (fgaz) [reply pkgs/games/gnonograms/default.nix:30]: Oh? I thought wrapGAppsHook took care of that, the program does work for me.

I'll re-enable the script then.

Comment (jtojnar) [reply pkgs/games/gnonograms/default.nix:30]: wrapGAppsHook only takes care of wrapping. Schema compilation is still responsibility of the package. This is discussed in https://github.com/NixOS/nixpkgs/issues/1455. Though, if you install it through environment.systemPackages, it [might be compiled by NixOS](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/nixos/modules/config/system-path.nix#L170-L172).

Comment (jtojnar) [reply pkgs/games/gnonograms/default.nix:30]: Looks like unlike most apps, the app just gracefully handles the cases when schema is not installed:

https://github.com/jeremypw/gnonograms/blob/18544494ef007392e36bb4e295500cd31ad4e70a/src/Controller.vala#L89-L91

Comment (fgaz) [reply pkgs/games/gnonograms/default.nix]: Thanks @jtojnar, this is good to know.
It should be fixed now.

Also wow, gsettings is _very_ complex..

**Comment by fgaz** [comment]
Oh. Aren't new packages backportable?

**Pull request merged**